tests/mmu_file_processor_test.py

```diff
@@ -0,0 +1,62 @@
+import unittest
+import os
+import shutil
+
+from extras.mmu_file_processor import MmuFileProcessor
+
+class TestMmuFileProcessor(unittest.TestCase):
```

Comment:
I added some unit tests because testing on various real-world files was getting tedious

Reply:
I really like this because I will use it!!   I haven't had a chance to try yet but hopefully will tomorrow.
To be clear this will only work if the entire file is immediately available so it should probably be noted that the few (only encountered 2) folks that use Octoprint and stream the gcode will be out of luck.  We just need to make sure that logic protects against this case and does nothing...